Firestore/core/src/firebase/firestore/auth/firebase_credentials_provider_apple.mm

```diff
@@ -31,8 +33,9 @@
 namespace firestore {
 namespace auth {
 
-FirebaseCredentialsProvider::FirebaseCredentialsProvider(FIRApp* app)
-    : contents_(std::make_shared<Contents>(app, User::FromUid([app getUID]))) {
+FirebaseCredentialsProvider::FirebaseCredentialsProvider(FIRApp* app) {
+  id<FIRAuthInterop> auth = FIR_COMPONENT(FIRAuthInterop, app.container);
```

Comment:
The amount of boilerplate in the tests here makes me think that we should pull the interaction with the container out of this class.

As a general principle we should isolate code from the dependency injector and I think this principle should apply all the way out to FIRFirestore. By the time you've constructed a FIRFirestore instance we should be able to supply it with all the dependencies it needs. 

Please change this constructor to take `FIRApp* app, id<FIRAuthInterop> auth`.